<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1366</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* Groupbox */

QGroupBox {
    border: 1px solid #DDD;
    border-radius: 9px;
    margin-top: 9px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>6</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item>
     <widget class="QGroupBox" name="SettingsGB">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Settings</string>
      </property>
      <widget class="QLabel" name="RotationLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>110</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Rotation around Y-axis</string>
       </property>
      </widget>
      <widget class="QDial" name="RotationDial">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>130</y>
         <width>50</width>
         <height>64</height>
        </rect>
       </property>
       <property name="maximum">
        <number>359</number>
       </property>
       <property name="wrapping">
        <bool>true</bool>
       </property>
       <property name="notchTarget">
        <double>12.000000000000000</double>
       </property>
       <property name="notchesVisible">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QSpinBox" name="SubdivSteps">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>260</y>
         <width>181</width>
         <height>29</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="ImportOBJ">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>40</y>
         <width>181</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>Import OBJ file</string>
       </property>
      </widget>
      <widget class="QLabel" name="StepsLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>230</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Catmull-Clark steps</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="MoveToLimPos">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>310</y>
         <width>201</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string> Move vertices to their limit position</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="TessMode">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>380</y>
         <width>201</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Render bezier patches...</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="Od">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>550</y>
         <width>41</width>
         <height>22</height>
        </rect>
       </property>
       <property name="decimals">
        <number>1</number>
       </property>
       <property name="minimum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.200000000000000</double>
       </property>
       <property name="value">
        <double>3.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="Ob">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>640</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="decimals">
        <number>1</number>
       </property>
       <property name="minimum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.200000000000000</double>
       </property>
       <property name="value">
        <double>3.000000000000000</double>
       </property>
      </widget>
      <widget class="QFrame" name="frame">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>470</y>
         <width>201</width>
         <height>181</height>
        </rect>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="lineWidth">
        <number>3</number>
       </property>
       <widget class="QFrame" name="frame_3">
        <property name="geometry">
         <rect>
          <x>50</x>
          <y>50</y>
          <width>91</width>
          <height>81</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="lineWidth">
         <number>3</number>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="I1">
        <property name="geometry">
         <rect>
          <x>60</x>
          <y>40</y>
          <width>71</width>
          <height>22</height>
         </rect>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="minimum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.200000000000000</double>
        </property>
        <property name="value">
         <double>3.000000000000000</double>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="I2">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>80</y>
          <width>41</width>
          <height>22</height>
         </rect>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="minimum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.200000000000000</double>
        </property>
        <property name="value">
         <double>3.000000000000000</double>
        </property>
       </widget>
      </widget>
      <widget class="QDoubleSpinBox" name="Oa">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>550</y>
         <width>41</width>
         <height>22</height>
        </rect>
       </property>
       <property name="decimals">
        <number>1</number>
       </property>
       <property name="minimum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.200000000000000</double>
       </property>
       <property name="value">
        <double>3.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="Oc">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>460</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="decimals">
        <number>1</number>
       </property>
       <property name="minimum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.200000000000000</double>
       </property>
       <property name="value">
        <double>3.000000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="RotationLabel_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>420</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>... and use tessellation levels:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>660</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>(0,0)</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>660</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>(1,0)</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>450</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>(0,1)</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>450</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>(1,1)</string>
       </property>
      </widget>
      <zorder>RotationLabel</zorder>
      <zorder>RotationDial</zorder>
      <zorder>SubdivSteps</zorder>
      <zorder>ImportOBJ</zorder>
      <zorder>StepsLabel</zorder>
      <zorder>MoveToLimPos</zorder>
      <zorder>TessMode</zorder>
      <zorder>frame</zorder>
      <zorder>Oa</zorder>
      <zorder>Oc</zorder>
      <zorder>Od</zorder>
      <zorder>Ob</zorder>
      <zorder>RotationLabel_2</zorder>
      <zorder>label</zorder>
      <zorder>label_2</zorder>
      <zorder>label_3</zorder>
      <zorder>label_4</zorder>
     </widget>
    </item>
    <item>
     <widget class="MainView" name="MainDisplay"/>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MainView</class>
   <extends>QOpenGLWidget</extends>
   <header>mainview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
